Trident/offsetfinder.c

```diff
@@ -0,0 +1,324 @@
+//
+//  offsetfinder.c
+//  Trident
+//
+//  Created by Benjamin Randazzo on 14/12/2016.
+//  Copyright © 2016 Benjamin Randazzo. All rights reserved.
+//
+
+#include "offsetfinder.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#define determineTarget(modelIdentifier, swVers, target) if (!strcmp(device_model, modelIdentifier)) { if (!strcmp(system_version, swVers)) { return target; }}
+
+t_target_environment target_environment = NotSupported;
+
+t_target_environment info_to_target_environment(const char *device_model, const char *system_version) {
+	determineTarget("iPhone5,2", "9.2.1", iPhone52_iOS921);
+	determineTarget("iPhone4,1", "9.3.1", iPhone41_iOS931);
+	determineTarget("iPhone4,1", "9.3.2", iPhone41_iOS932);
+	determineTarget("iPhone5,2", "9.3.2", iPhone52_iOS932);
+	determineTarget("iPhone5,3", "9.3.2", iPhone53_iOS932);
+	determineTarget("iPad2,1", "9.3.2", iPad21_iOS932);
+	determineTarget("iPad2,2", "9.3.2", iPad22_iOS932);
+	determineTarget("iPad2,3", "9.3.2", iPad23_iOS932);
+	determineTarget("iPad2,4", "9.3.2", iPad24_iOS932);
+	determineTarget("iPad2,3", "9.3.3", iPad23_iOS933);
+	determineTarget("iPad3,1", "9.3.4", iPad31_iOS934);
+	return NotSupported;
+}
+
+void init_target_environment(const char *device_model, const char *system_version) {
+	target_environment = info_to_target_environment(device_model, system_version);
+}
+
+uint32_t find_OSSerializer_serialize(void) {
+	switch (target_environment) {
+		case iPhone52_iOS921: return 0x317868;
+		case iPhone41_iOS931: return 0x31812c;
+		case iPhone41_iOS932: return 0x318264;
+		case iPhone52_iOS932: return 0x31ef58;
+		case iPhone53_iOS932: return 0x31ef58;
+		case iPad21_iOS932: return 0x318264;
```

Comment:
Hi, I've added support for the iPhone 4S iOS 9.3.2 in the refactored abstraction method you've added. Should I make a pull request, or are you still working on the refactored abstraction method?